### PR TITLE
Katya: PeerDAS Metrics project proposal 

### PR DESCRIPTION
**Pull request by KatyaRyazantseva**
Ekaterina Riazantseva: The proposal of the fellowship project PeerDAS Metrics

### CODE REVIEW

File: projects/peerdas-metrics.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+Improving PeerDAS testing efficiency via metrics implementation and visualization. 
+
+## Motivation
+
+PeerDAS is now entering its testing phase, with most CL clients actively participating. Identifying and utilizing specific metrics is crucial for evaluating efficiency, troubleshooting, and optimizing the protocol. Before PeerDAS is deployed on the mainnet, detailed testing is important to keep the Ethereum Protocol secure and efficient. The project aims to analyze and present these results in an accessible format for core developers and the research team.
+
+## Project description
+
+PeerDAS Metrics focuses on identifying the most crucial metrics, integrating them into CL clients (Teku, Grandine), and developing visual representations to help teams use these metrics effectively. Additionally, the project will present the analysis of these metrics, providing valuable insights for the research team.
```
Comment (taxmeifyoucan) [inline projects/peerdas-metrics.md:11]: Does this mean creating a new metrics category in clients? I see teku collects 'network', 'libp2p' categories for example, I guess there might be an overlap but I am not sure which one is for blobs. I wonder if you are creating 'peerdas' metrics namespace or just improving existing

File: projects/peerdas-metrics.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+## Specification
+
+The project is executed through these specific steps:
+
+- Monitor and visualize the initial set of metrics using Kurtosis and Grafana.
```
Comment (taxmeifyoucan) [inline projects/peerdas-metrics.md:17]: I guess with prometheus as a part of the stack?

File: projects/peerdas-metrics.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+- Review the initial metrics, identify additional ones that could be beneficial.
```
Comment (taxmeifyoucan) [inline projects/peerdas-metrics.md:19]: Lighthouse integrated metrics for peerdas development in its branch, Andrew listed it in the r&d discord message. My idea was to get inspiration from that and apply it in other clients. If so, the actual list of metrics or at least vaguely categorized should be here

File: projects/peerdas-metrics.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+- Integrate these metrics into Teku and potentially into Grandine.
+- Discuss and potentially develop APIs based on the metrics.
```
Comment (taxmeifyoucan) [inline projects/peerdas-metrics.md:21]: That would mean new Beacon API?

File: projects/peerdas-metrics.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+## Motivation
+
+PeerDAS is now entering its testing phase, with most CL clients actively participating. Identifying and utilizing specific metrics is crucial for evaluating efficiency, troubleshooting, and optimizing the protocol. Before PeerDAS is deployed on the mainnet, detailed testing is important to keep the Ethereum Protocol secure and efficient. The project aims to analyze and present these results in an accessible format for core developers and the research team.
+
+## Project description
+
+PeerDAS Metrics focuses on identifying the most crucial metrics, integrating them into CL clients (Teku, Grandine), and developing visual representations to help teams use these metrics effectively. Additionally, the project will present the analysis of these metrics, providing valuable insights for the research team.
+
+## Specification
```
Comment (taxmeifyoucan) [inline projects/peerdas-metrics.md:13]: This list isn't really a specification but suits more the roadmap part. It should be a technical description of your work with links to details. If it's not ready, you can just keep it like this and update later

File: projects/peerdas-metrics.md (hunk 1 above)
Comment (KatyaRyazantseva) [reply projects/peerdas-metrics.md:11]: Every client implements metrics in its own way. Some of these metrics are being added into existing categories. Others were put into new categories. So, it's a bit tricky to categorize them in the similar manner. That's what I am going to find out. The main goal is to get the results from these metrics and to analyze them. I consider it to be helpful for monitoring PeerDAS issues.

File: projects/peerdas-metrics.md (hunk 5 above)
Comment (KatyaRyazantseva) [reply projects/peerdas-metrics.md:13]: I am going to improve it as far as I am moving forward. Will keep it in mind, thank you

**Pull request merged**